### PR TITLE
Problem: MLS member leaving/joining commit success not validated (WIP #341)

### PR DESCRIPTION
**Pull request by tomtau**
Solution: sketched out the NACK submission based on the MLS protocol discussions

### CODE REVIEW

File: docs/modules/tdbe.md [hunk 1]
```diff
@@ -170,9 +172,19 @@ pub struct SelfUpdateProposal {
...
+    nack: Vec<u8>, // msg ref + `zz` + DLEQ proof -- TBD: https://github.com/mlswg/mls-protocol/issues/21#issuecomment-455392023
+}
+```
+
+(TODO: `CommitChangeTx` / `SelfUpdateProposal` may need to contain Schnorr NIZK of knowledge (RFC 8235)
+for encrypted parts (`Welcome`, `DirectPathNode`...))
 
 ##### TDBE handling / generation
 TDBE runs as a standalone enclave which includes TM light client and reacts to information received from it.
+For fetched transactions with MLS hanshake messages, it should verify they are valid (i.e. their TXID is a leaf in the Merkle tree
```
Comment (lezzokafka) [inline docs/modules/tdbe.md]:
```suggestion
For fetched transactions with MLS handshake messages, it should verify they are valid (i.e. their TXID is a leaf in the Merkle tree
```

File: docs/modules/tdbe.md [hunk 2]
```diff
@@ -229,19 +255,12 @@ Notably, it needs:
...
-if not, there are two possibilities
-  - local node problem (e.g. no running TDBE)
-  - invalid Commit (e.g. DirectPath)
-In any case, it should punish the proposer/submitter of that `CommitChangeTx`
-(if chain-abci should know identity of its staking address / TDBE and self-shutdown if it originated from its "node")
-and expect to receive `CommitChangeTx` from the next leaf.
-
-[[ OPEN ISSUE: 
+- to keep track if valid `NACK` was received after a "valid" `CommitChangeTx` -- NACK invalidating `CommitChangeTx` should treat the `CommitChangeTx` similar to byzantine faults
+- after CommitChangeTx -- after block commit and NACK timeout, enquire TVE if it was pushed a new key;
```
Comment (lezzokafka) [inline docs/modules/tdbe.md]:
```suggestion
- after `CommitChangeTx` -- after block commit and NACK timeout, enquire TVE if it was pushed a new key;
```

**Review by lezzokafka — APPROVED** [review]
Looks great, may I ask did we mentioned _HPKE_ somewhere in the doc?

**Comment by tomtau** [comment]
> Looks great, may I ask did we mentioned _HPKE_ somewhere in the doc?

it's in https://github.com/mlswg/mls-protocol/blob/master/draft-ietf-mls-protocol.md ?

**Review by lezzokafka — APPROVED** [review]

**Review by devashishdxt — APPROVED** [review]

**Pull request merged**